ZKSyncSDK/Classes/Wallet/DefaultWallet+ChangePubKey.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  DefaultWallet+ChangePubKey.swift
+//  ZKSyncSDK
+//
+//  Created by Eugene Belyakov on 16/01/2021.
+//
+
+import Foundation
+import PromiseKit
+import BigInt
+
+extension DefaultWallet {
+
+    public func setSigningKey(fee: TransactionFee, nonce: Int32?, oncahinAuth: Bool, completion: @escaping (Swift.Result<String, Error>) -> Void) {
+        
+        guard !isSigningKeySet else {
+            completion(.failure(WalletError.signingKeyAlreadySet))
+            return
+        }
+        
+        firstly {
+            return nonce != nil ? .value(nonce!) : getNonce()
+        }.then { nonce in
+            self.buildSignedChangePubKeyTx(fee: fee, nonce: nonce, onchainAuth: oncahinAuth)
+        }.then { signedTransaction in
+            self.submitSignedTransaction(signedTransaction.transaction,
+                                         ethereumSignature: signedTransaction.ethereumSignature,
+                                         fastProcessing: false)
+        }.pipe { result in
+            completion(result.result)
+        }
+    }
+    
+    func buildSignedChangePubKeyTx(fee: TransactionFee,
+                                   nonce: Int32,
+                                   onchainAuth: Bool) -> Promise<SignedTransaction<ChangePubKey>> {
+        return firstly {
+            getTokens()
```

Comment:
If it is feasible, the tokens should be cached so that the users don't have to fetch them each time they need to do some formatting with tokens or get a token's id

Reply:
Implemented caching of `Tokens` inside `DefaultProvider` implementation